test/server/rules.bzl

```diff
@@ -15,30 +15,23 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
-def grakn_test(grakn_artifact = None,
-               deps = [],
-               classpath_resources = [],
-               data = [],
-               **kwargs):
+def grakn_java_test(name, grakn_artifact_linux, grakn_artifact_mac, deps = [], classpath_resources = [], data = [], **kwargs):
 
-    new_data = [] + data
-
-    if grakn_artifact != None:
-        new_data += [grakn_artifact]
-
-    location = "$(location {})".format(grakn_artifact) if grakn_artifact != None else "none"
+    native.genrule(
+        name = "grakn_artifact_selector",
+        outs = ["grakn-core-server.tar.gz"],
+        srcs = select({
+            "@graknlabs_dependencies//util/platform:is_mac": [grakn_artifact_mac],
+            "@graknlabs_dependencies//util/platform:is_linux": [grakn_artifact_linux],
+        }, no_match_error = "There is no Grakn Core artifact compatible with this operating system. Supported operating systems are Mac and Linux."),
+        cmd = "read -a srcs <<< '$(SRCS)' && read -a outs <<< '$(OUTS)' && cp $${srcs[0]} $${outs[0]}",
+    )
 
     native.java_test(
-        deps = depset(deps + [
-            "@graknlabs_common//test/server:grakn-setup",
-        ]).to_list(),
-        classpath_resources = depset(classpath_resources + [
-            "@graknlabs_common//test/server:logback",
-        ]).to_list(),
-        data = depset(new_data).to_list(),
-        args = [
-            location,
-        ],
+        name = name,
+        deps = depset(deps + ["@graknlabs_common//test/server:grakn-setup"]).to_list(),
+        classpath_resources = depset(classpath_resources + ["@graknlabs_common//test/server:logback"]).to_list(),
+        data = depset(data + ["grakn-core-server.tar.gz"]).to_list(),
```

Comment:
`"grakn-core-server.tar.gz"` looks weird - its the output of a bazel target, not a bazel target... 

In this case, we have both options -- however its more idiomatic to use the target I think. It should work because there is only 1 output file of the target. 

Referring to an outputted file by name will be very useful if a rule outputs multiple files.

what do you think?

Reply:
Oh I didn't know that was possible with a `genrule` - will test it now

Reply:
yaaas